File: scala-pr/54/scala/code-quality-error-handling/JsonParser.scala
import scala.util.parsing.json.JSON

class JsonParser {
  def parseJson(jsonString: String): Any = {
    // {fact rule=code-quality-error-handling@v1.0 defects=1}
    try {
      JSON.parseFull(jsonString).getOrElse(null)
    } catch {
      case _: Exception => throw new Exception("Something went wrong") // Generic exception
    }
    // {/fact}
  }
}

File: kotlin-pr/50/kotlin/code-quality-error-handling/JsonParser.kt
import kotlinx.serialization.json.Json

class JsonParser {
    fun parseJson(jsonString: String): Any? {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        return try {
            Json.parseToJsonElement(jsonString)
        } catch (e: Exception) {
            throw Exception("Something went wrong") // Generic exception
        }
        // {/fact}
    }
}